File: backend/src/main/java/com/example/backend/service/ProteinService.java
package com.example.backend.service;

import com.example.backend.model.Protein;
import com.example.backend.model.exceptions.ProteinDoesNotExist;
import uk.ac.ebi.uniprot.dataservice.client.exception.ServiceException;

import java.util.List;
import java.util.Optional;

public interface ProteinService {
    Optional<Protein> getDetailsAboutProtein(String name) throws ServiceException, ProteinDoesNotExist;
    String getProteinAbstract();
}

File: frontend/src/services/ProteinService.js
import axios from "../custom-axios/axios"

const ProteinService={
    getProteinAbstract:()=>{
        return axios.get('/proteins')
    },
    getDetailsAboutProtein:(name)=>{
        return axios.get(`/proteins/${name}`)
    }
}
export default ProteinService;